loader/loader_test.go

```diff
@@ -0,0 +1,282 @@
+	})
+
+	It("ensures environments can be found", func() {
+		environments, err := GetAllEnvironments(k8sClient, ctx, hasApp)
+		Expect(err).To(BeNil())
+		Expect(environments).NotTo(BeNil())
+	})
+
+	It("ensures all Releases exists when HACBSTests succeeded", func() {
+		Expect(k8sClient).NotTo(BeNil())
+		Expect(ctx).NotTo(BeNil())
+		Expect(hasSnapshot).NotTo(BeNil())
+		gitops.MarkSnapshotAsPassed(k8sClient, ctx, hasSnapshot, "test passed")
+		Expect(gitops.HaveHACBSTestsSucceeded(hasSnapshot)).To(BeTrue())
+
+		// Normally we would Ensure that releases exist here, but that requires
+		// importing the snapshot package which causes an import cycle
+
+		releases, err := GetReleasesWithSnapshot(k8sClient, ctx, hasSnapshot)
+		Expect(err).To(BeNil())
+		Expect(releases).NotTo(BeNil())
+		for _, release := range *releases {
+			Expect(k8sClient.Delete(ctx, &release)).Should(Succeed())
+		}
+	})
+
+	It("ensures the Application Components can be found ", func() {
+		applicationComponents, err := GetAllApplicationComponents(k8sClient, ctx, hasApp)
+		Expect(err).To(BeNil())
+		Expect(applicationComponents).NotTo(BeNil())
+	})
+
+	It("ensures we can get an Application from a Snapshot ", func() {
+		app, err := GetApplicationFromSnapshot(k8sClient, ctx, hasSnapshot)
+		Expect(err).To(BeNil())
+		Expect(app).NotTo(BeNil())
+		Expect(app.ObjectMeta).To(Equal(hasApp.ObjectMeta))
+	})
+
+	It("ensures we can get a Component from a Snapshot ", func() {
+		comp, err := GetComponentFromSnapshot(k8sClient, ctx, hasSnapshot)
+		Expect(err).To(BeNil())
+		Expect(comp).NotTo(BeNil())
+		Expect(comp.ObjectMeta).To(Equal(hasComp.ObjectMeta))
+	})
+
+	It("ensures we can get a Component from a Pipeline Run ", func() {
+		comp, err := GetComponentFromPipelineRun(k8sClient, ctx, testPipelineRun)
+		Expect(err).To(BeNil())
+		Expect(comp).NotTo(BeNil())
+		Expect(comp.ObjectMeta).To(Equal(hasComp.ObjectMeta))
+	})
+
+	It("ensures we can get the application from the Pipeline Run", func() {
+		app, err := GetApplicationFromPipelineRun(k8sClient, ctx, testPipelineRun)
+		Expect(err).To(BeNil())
+		Expect(app).NotTo(BeNil())
+		Expect(app.ObjectMeta).To(Equal(hasApp.ObjectMeta))
+	})
+
+	It("ensures we can get the environment from the Pipeline Run", func() {
+		env, err := GetApplicationFromPipelineRun(k8sClient, ctx, testPipelineRun)
+		Expect(err).To(BeNil())
+		Expect(env).NotTo(BeNil())
+	})
+
+	It("ensures we can get the Application from a Component", func() {
+		app, err := GetApplicationFromComponent(k8sClient, ctx, hasComp)
+		Expect(err).To(BeNil())
+		Expect(app).NotTo(BeNil())
+		Expect(app.ObjectMeta).To(Equal(hasApp.ObjectMeta))
+	})
+
+	It("ensures we can get the Snapshot from a Pipeline Run", func() {
+		snapshot, err := GetSnapshotFromPipelineRun(k8sClient, ctx, testPipelineRun)
+		Expect(err).To(BeNil())
+		Expect(snapshot).NotTo(BeNil())
+		Expect(snapshot.ObjectMeta).To(Equal(hasSnapshot.ObjectMeta))
+	})
```

Comment:
This is the test that is failing, with the error:
```
• [FAILED] [0.025 seconds]
Loader [It] ensures we can get the Snapshot from a Pipeline Run
/var/home/ryan/Development/RHTAP/integration-service/loader/loader_test.go:276

  [FAILED] Expected
      <*errors.errorString | 0xc000a8e690>: {
          s: "the pipeline has no snapshot associated with it",
      }
  to be nil
  In [It] at: /var/home/ryan/Development/RHTAP/integration-service/loader/loader_test.go:278 @ 04/26/23 15:45:10.287
------------------------------

Summarizing 1 Failure:
  [FAIL] Loader [It] ensures we can get the Snapshot from a Pipeline Run
  /var/home/ryan/Development/RHTAP/integration-service/loader/loader_test.go:278
```

How can I add a snapshot to the pipeline run?

Reply:
That worked, thanks for your help.